user_tools/src/spark_rapids_tools/tools/qualification_stats_report.py

```diff
@@ -0,0 +1,99 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Implementation of the Qualification Stats Report."""
+
+
+from dataclasses import dataclass, field
+from logging import Logger
+
+import pandas as pd
+import fire
+
+from spark_rapids_pytools.common.utilities import ToolLogging
+
+
+@dataclass
+class SparkQualificationStats:
+    """
+    Encapsulates the logic to generate the Qualification Stats Report.
+    """
+    logger: Logger = field(default=None, init=False)
+    output_file: str
+    unsupported_operators_df: pd.DataFrame
+    stages_df: pd.DataFrame
+    output_columns: dict
+    result_df: pd.DataFrame = field(default=None, init=False)
+
+    def __post_init__(self):
+        self.logger = ToolLogging.get_and_setup_logger('rapids.tools.qualification.stats')
+
+    def read_dataframes(self):
```

Comment:
nit: This function does not seem to read DF anymore. Can we rename and update the exception msg?